src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -596,28 +594,10 @@ impl Step for Rustdoc {
             && target_compiler.stage > 0
             && builder.rust_info().is_managed_git_subrepository()
         {
-            let commit = get_closest_merge_commit(
-                Some(&builder.config.src),
-                &builder.config.git_config(),
-                &[],
-            )
-            .unwrap();
-
-            let librustdoc_src = builder.config.src.join("src/librustdoc");
-            let rustdoc_src = builder.config.src.join("src/tools/rustdoc");
-
-            // FIXME: The change detection logic here is quite similar to `Config::download_ci_rustc_commit`.
-            // It would be better to unify them.
-            let has_changes = !git(Some(&builder.config.src))
-                .allow_failure()
-                .run_always()
-                .args(["diff-index", "--quiet", &commit])
-                .arg("--")
-                .arg(librustdoc_src)
-                .arg(rustdoc_src)
-                .run(builder);
-
-            if !has_changes {
+            let files_to_track = &["src/librustdoc", "src/tools/rustdoc"];
+
+            // Check if unchanged
+            if builder.config.last_modified_commit(files_to_track, "rustdoc", true).is_some() {
```

Comment:
I missed this "rustdoc" value here sorry; it should be "download-rustc" instead.

Reply:
why it shoud be that

Reply:
I noticed `option_name` appears as logs

Reply:
Yes, and these:

https://github.com/rust-lang/rust/blob/9abfcb49002b82f46ad66ceb0b26cee8be2dc658/src/bootstrap/src/core/config/config.rs#L2878

https://github.com/rust-lang/rust/blob/9abfcb49002b82f46ad66ceb0b26cee8be2dc658/src/bootstrap/src/core/config/config.rs#L2898-L2901

https://github.com/rust-lang/rust/blob/9abfcb49002b82f46ad66ceb0b26cee8be2dc658/src/bootstrap/src/core/config/config.rs#L2905-L2907

don't make any sense with invalid option name.

Reply:
> This logic is gated from "download-rustc" option and there is no option such "rustdoc" in config.toml.
Ah, I haven't thought that far ahead; not  thought about connecting these code changes to `config.toml yet`. To be honest, I'm not sure how they're linked.

but I will change it for now